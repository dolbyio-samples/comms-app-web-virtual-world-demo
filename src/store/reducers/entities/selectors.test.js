import {
  getEntities,
  getEntity,
  getValidEntities,
  getValueFrom,
} from "./selectors";
import { IsLoading, NotFoundEntity } from "../../../dataDefinitions/defect";
import { describe, it, expect } from "vitest";

describe("selectors it( suite", () => {
  it("should return isLoading", () => {
    // 1- arrange
    const state = {};
    const entityType = "type";
    const id = "loading";

    // 2- act
    const entity = getEntity(state, entityType, id);

    // 3- assert
    expect(entity).toBe(IsLoading);
  });

  it("should return NotFoundEntity", () => {
    // 1- arrange
    const state = {
      entities: {
        firstType: {
          1: { prop1: 1 },
        },
      },
    };
    const entityType = "firstType";
    const id = "2";

    // 2- act
    const entity = getEntity(state, entityType, id);

    // 3- assert
    expect(entity).toBe(NotFoundEntity);
  });

  it("should return selected entity", () => {
    // 1- arrange
    const state = {
      entities: {
        firstType: {
          1: { prop1: 1 },
        },
      },
    };
    const entityType = "firstType";
    const id = "1";

    // 2- act
    const entity = getEntity(state, entityType, id);

    // 3- assert
    expect(entity).toMatchSnapshot();
  });

  it("should return an empty array", () => {
    // 1- arrange
    const state = {
      entities: {
        firstType: {
          1: { prop1: 1 },
          2: { prop1: 2 },
          3: { prop1: 2 },
          4: { prop1: 2 },
        },
      },
    };
    const entityType = "firstType";

    // 2- act
    const entities = getEntities(state, entityType, 1);

    // 3- assert
    expect(entities).toEqual([]);
  });

  it("should return an array of selected entities", () => {
    // 1- arrange
    const state = {
      entities: {
        firstType: {
          1: { prop1: 1 },
          2: { prop1: 2 },
          3: { prop1: 2 },
          4: { prop1: 2 },
        },
      },
    };
    const entityType = "firstType";
    const ids = ["1", "3"];

    // 2- act
    const entities = getEntities(state, entityType, ids);

    // 3- assert
    expect(entities).toMatchSnapshot();
  });

  it("should return an array of selected entities", () => {
    // 1- arrange
    const state = {
      entities: {
        firstType: {
          1: { prop1: 1 },
          2: { prop1: 2 },
          3: { prop1: 2 },
          4: { prop1: 2 },
        },
      },
    };
    const entityType = "firstType";
    const ids = ["1", "3"];

    // 2- act
    const entities = getValidEntities(state, entityType, ids);

    // 3- assert
    expect(entities).toMatchSnapshot();
  });

  it("should return 1", () => {
    // arrange
    const object = {
      prop1: 1,
    };

    // act
    const value = getValueFrom(object, "prop1");

    // assert
    expect(value).toBe(1);
  });

  it("should return NotFoundEntity", () => {
    // arrange
    const object = {
      prop1: 1,
    };

    // act
    const value = getValueFrom(object, "prop2");

    // assert
    expect(value).toBe(NotFoundEntity);
  });
});
