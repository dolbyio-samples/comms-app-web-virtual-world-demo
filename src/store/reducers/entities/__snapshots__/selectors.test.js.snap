// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`selectors it( suite > should return an array of selected entities 1`] = `
[
  {
    "prop1": 1,
  },
  {
    "prop1": 2,
  },
]
`;

exports[`selectors it( suite > should return an array of selected entities 2`] = `
[
  {
    "prop1": 1,
  },
  {
    "prop1": 2,
  },
]
`;

exports[`selectors it( suite > should return selected entity 1`] = `
{
  "prop1": 1,
}
`;
